models/src/main/scala/coop/rchain/models/SortedParMap.scala

```diff
@@ -0,0 +1,28 @@
+package coop.rchain.models
+
+import scala.collection.MapLike
+import coop.rchain.models.rholang.sort.ordering._
+
+case class SortedParMap(private val ps: Map[Par, Par])
+    extends Map[Par, Par]
+    with MapLike[Par, Par, SortedParMap] {
+
+  lazy val sortedMap: List[(Par, Par)] = ps.sort
+
+  override def empty: SortedParMap = SortedParMap.empty
+
+  override def get(key: Par): Option[Par] = ps.get(key)
+
+  override def iterator: Iterator[(Par, Par)] = sortedMap.toIterator
+
+  override def +[V1 >: Par](kv: (Par, V1)): Map[Par, V1] = sortedMap.toMap + kv
```

Comment:
This is weird,  method removing an element (`-`) returns `This` (`SortedParMap`) but one that adds (`+`) returns `Map[K, V1 <: V]` but this is the way `MapLike` interface requires.

Reply:
That's fine.